src/GafferArnold/IECoreArnoldPreview/Renderer.cpp

```diff
@@ -678,6 +680,53 @@ class ShaderCache : public IECore::RefCounted
 
 IE_CORE_DECLAREPTR( ShaderCache )
 
+class NodeVectorCache : public IECore::RefCounted
```

Comment:
I think generalising this as a `NodeVectorCache` might be misleading. We're relying entirely on the fact that lights are added/removed/updated prior to objects, as otherwise the contents of our cache can contain totally invalid pointers. Generalising it encourages folks to think it can be used for other stuff where this ordering doesn't hold, which could lead to bugs. Since we don't have another use case anything, I suggest we just call it LightListCache and remove the prefix argument. We should probably also mention our reliance on the order of operations in a comment somewhere...

Reply:
I've merged the changes into 16470e0. It all makes sense, thanks :)